src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.component.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useEffect, useState } from 'react'
+import { elementDidMount, useHeightObserver } from '../../../../../../common'
+import CopyToast, { handleSelectionChange } from '../CopyToast'
+import { Terminal } from 'xterm'
+import { FitAddon } from 'xterm-addon-fit'
+import * as XtermWebfont from 'xterm-webfont'
+import SockJS from 'sockjs-client'
+import { SocketConnectionType } from '../node.type'
+import IndexStore from '../../../../index.store'
+import { AppType } from '../../../../appDetails.type'
+import moment from 'moment'
+import { CLUSTER_STATUS } from '../../../../../../ClusterNodes/constants'
+import { TERMINAL_STATUS } from './constants'
+import './terminal.scss'
+import { TerminalViewType } from './terminal.type'
+
+let socket 
+let terminal 
+let fitAddon 
+let clusterTimeOut
+
+export default function TerminalView({
+    terminalRef,
+    sessionId,
+    socketConnection,
+    setSocketConnection,
+    isTerminalTab = true,
+    renderConnectionStrip,
+    registerLinkMatcher,
+    terminalMessageData,
+    clearTerminal,
+}: TerminalViewType) {
+    const [firstMessageReceived, setFirstMessageReceived] = useState(false)
+    const [isReconnection, setIsReconnection] = useState(false)
+    const appDetails = IndexStore.getAppDetails()
+    const [popupText, setPopupText] = useState<boolean>(false)
+
+    function resizeSocket() {
+        if (terminal && fitAddon && isTerminalTab) {
+            const dim = fitAddon.proposeDimensions()
+            if (dim && socket?.readyState === WebSocket.OPEN) {
+                socket?.send(JSON.stringify({ Op: 'resize', Cols: dim.cols, Rows: dim.rows }))
+            }
+            fitAddon.fit()
+        }
+    }
+
+    const [myDivRef] = useHeightObserver(resizeSocket)
+
+    useEffect(() => {
+        if (!terminal) {
+            elementDidMount('#terminal-id').then(() => {
+                createNewTerminal()
+            })
+        }
+        if (sessionId && terminal) {
+            setIsReconnection(true)
+            postInitialize(sessionId)
+        } else {
+            setSocketConnection(SocketConnectionType.DISCONNECTED)
+        }
+    }, [sessionId])
+
+    useEffect(() => {
+        if (popupText){
+            setTimeout(() => setPopupText(false), 2000)
+        }
+    }, [popupText])
+
+    useEffect(() => {
+        if (socketConnection === SocketConnectionType.DISCONNECTING) {
+            if (clusterTimeOut) {
+                clearTimeout(clusterTimeOut)
+            }
+            if (socket) {
+                socket.close()
+                socket = undefined
+            }
+        }
+    }, [socketConnection])
+
+    const createNewTerminal = () => {
+        terminal = new Terminal({
+            scrollback: 99999,
+            fontSize: 14,
+            lineHeight: 1.4,
+            cursorBlink: false,
+            fontFamily: 'Inconsolata',
+            screenReaderMode: true,
+            theme: {
+                background: '#0B0F22',
```

Comment:
we should use color name from color palatte here

Reply:
Its not possible to pass formula